torchvision/prototype/transforms/_type_conversion.py

```diff
@@ -33,3 +37,31 @@ def extra_repr(self) -> str:
             return ""
 
         return f"num_categories={self.num_categories}"
+
+
+class ToTensor(Transform):
```

Comment:
The only benefit of using `ToTensor` over `PILToTensor` is that the former also handles `numpy` images, i.e. arrays of shape `H x W x C`. So far, we haven't talked about such images. Do we want to keep supporting them?

Reply:
ToTensor is going to be deprecated

Reply:
> ToTensor is going to be deprecated.

Is that the case? So far I'm only aware of discussions that we might want to do this, but not definitive plan. If we go for a deprecation, doing it in the course of adapting the new API is good time though.

> About transforming numpy to tensor, we may create `NupmyToTensor` ?

Do we really want to have a separate transform for each input type? IMO the largest appeal of `ToTensor` for users is that they don't have to care about the input type and can use it in every scenario. My proposal would be to have only two transforms:

- `ToImageTensor`
- `ToImagePIL`

Both transforms take "arbitrary" inputs and convert them to either a `features.Image` or a `PIL.Image.Image`. The naming is closely aligned with our new kernels and they give the same convenience as before.